src/App.tsx

```diff
@@ -1,24 +1,214 @@
+
+  const visibleTodos = useMemo(() => {
+    return (todos.filter((todo) => {
+      switch (filterBy) {
+        case SortType.ALL:
+          return todo;
+
+        case SortType.ACTIVE:
+          return !todo.completed;
+
+        case SortType.COMPLETED:
+          return todo.completed;
+
+        default:
+          return true;
+      }
+    })
+    );
+  }, [filterBy, todos]);
+
+  const hasCompleteTodos = useMemo(() => {
+    return todos.some((todo) => todo.completed);
+  }, [todos]);
+
+  const mustBeCompleted = todos.filter((todo) => !todo.completed).length;
+
+  const loadTodosData = async () => {
+    try {
+      const todosFromServer = await getTodos(USER_ID);
+
+      setTempTodo(null);
+      setisInputDisable(false);
+      setTodos(todosFromServer);
+    } catch (error) {
+      setErrorMessage("Can't load data...");
+    }
+  };
+
+  useEffect(() => {
+    loadTodosData();
+  }, []);
+
+  const createNewTodo = (query: string) => {
+    const newId = Math.max(...todos.map((todo) => todo.id + 1));
+
+    const newTodo = {
+      id: newId,
+      userId: USER_ID,
+      title: query,
+      completed: false,
+    };
+
+    return newTodo;
+  };
+
+  const onCompletedChange = async (todo: Todo) => {
+    try {
+      await onUpdate(todo.id, {
+        id: todo.id,
+        userId: todo.userId,
+        title: todo.title,
+        completed: !todo.completed,
+      });
+
+      loadTodosData();
+      setTempTodo(null);
+    } catch (error) {
+      setErrorMessage('Unable to update a todo');
+    } finally {
+      setIsLoadingTodosIds([]);
+    }
+  };
+
+  const dataTodo = {
+    userId: USER_ID,
+    title,
+    completed: false,
+  };
+
+  const sendTodo = async () => {
+    setTempTodo({ id: 0, ...dataTodo });
+
+    try {
+      if (title) {
+        await addTodo(createNewTodo(title));
+      } else {
+        setErrorMessage('Empty title alowed');
+      }
+
+      loadTodosData();
+    } catch {
+      setTempTodo(null);
+      setErrorMessage('Unable to add a todo');
+      setisInputDisable(false);
+    }
+  };
+
+  const handleFormSubmit = (event: React.FormEvent) => {
+    setisInputDisable(true);
+
+    event.preventDefault();
+
+    sendTodo();
+    setTitle('');
+  };
+
+  const removeTodo = async (todoId: number) => {
+    setIsLoadingTodosIds(prevState => [...prevState, todoId]);
+
+    try {
+      await onDelete(todoId);
+      loadTodosData();
+    } catch (error) {
+      setTempTodo(null);
+      setErrorMessage('Unable to delete a todo');
+    } finally {
+      setIsLoadingTodosIds([]);
+    }
+  };
+
+  const toggleAllComlpleted = () => {
+    setIsLoadingTodosIds(visibleTodos.map((todo) => todo.id));
+
+    visibleTodos.map((todo) => onCompletedChange(todo));
+  };
+
+  const clearCompleted = async () => {
+    const clearCompletedTodos = todos.filter((todo) => todo.completed);
+
+    clearCompletedTodos.forEach((todo) => removeTodo(todo.id));
```

Comment:
use Promice.all and await all todos to be deleted, now it doesn`t show loader on clear comleted